Index: announcer/util/mail.py
===================================================================
--- announcer/util/mail.py	(revision 8666)
+++ announcer/util/mail.py	(working copy)
@@ -48,7 +48,8 @@
 def set_header(message, key, value, charset=None):
     if not charset:
         charset = message.get_charset() or 'ascii'
-    value = Header(value, charset, MAXHEADERLEN-(len(key)+2))
+    if key != 'Date':
+        value = Header(value, charset, MAXHEADERLEN-(len(key)+2))
     if message.has_key(key):
         message.replace_header(key, value)
     else:
