--- announcer/distributors/mail.py~	2010-10-05 11:20:13 +0300
+++ announcer/distributors/mail.py	2010-11-12 13:11:43 +0200
@@ -489,6 +489,7 @@
 
             alt_msg_format = 'html' in alternate_style and 'html' or 'plain'
             msgText = MIMEText(alternate_output, alt_msg_format)
+            msgText.set_charset(self._charset)
             parentMessage.attach(msgText)
         else:
             parentMessage = rootMessage
